hal/riscv64/sbi.c

```diff
@@ -126,6 +129,18 @@ void hal_sbiSetTimer(u64 stime)
 }
 
 
+__attribute__((noreturn)) void hal_sbiReset(u32 type, u32 reason)
+{
+	if (hal_sbiProbeExtension(SBI_EXT_SRST).error == 0) {
+		(void)hal_sbiEcall(SBI_EXT_SRST, SBI_SRST_RESET, type, reason, 0, 0, 0, 0);
+	}
+
+	/* Reset not available/failed, halt the system */
+	for (;;) {
+	}
```

Comment:
imho better to return and allow the caller to handle this (no need for retval, as fact of returning is an error indicator)